src/WSSlots.php

```diff
@@ -166,14 +220,40 @@ final public static function editSlots(
 		$flags = EDIT_INTERNAL;
 		$comment = CommentStoreComment::newUnsavedComment( $summary );
 
-		if ( $watchlist === "nochange" ) {
+		if ( $bot ) {
+			$flags |= EDIT_FORCE_BOT;
+		}
+
+		if ( $minor ) {
+			$flags |= EDIT_MINOR;
+		}
+
+		if ( $suppress ) {
 			$flags |= EDIT_SUPPRESS_RC;
 		}
 
 		$logger->debug( 'Calling saveRevision on PageUpdater' );
 		$pageUpdater->saveRevision( $comment, $flags );
 		$logger->debug( 'Finished calling saveRevision on PageUpdater' );
 
+		// Add the page to the watchlist
+		$watch = self::getWatchlistValue( $watchlist, $titleObject, $user );
+		if ( method_exists( MediaWikiServices::class, 'getWatchlistManager' ) ) {
+			// >1.37
+			MediaWikiServices::getInstance()->getWatchlistManager()->setWatch( $watch, $user, $titleObject );
+		} else {
+			// <=1.36
+			$watchedItemStore = MediaWikiServices::getInstance()->getWatchedItemStore();
+
+			if ( $watch ) {
+				$watchedItemStore->addWatch( $user, $titleObject );
+			} else {
+				$watchedItemStore->removeWatch( $user, $titleObject );
+			}
+
+			$user->invalidateCache();
```

Comment:
I think "invalidateCache" should not be called when `$watchlist=nochange`?
There is no change, no need to invalidate cache.

Or did I misunderstand what invalidatecache does? (I have no idea tbh)

Reply:
I copied this from core :)